src/main/java/org/opentripplanner/datastore/https/HttpsFileDataSource.java

```diff
@@ -0,0 +1,87 @@
+package org.opentripplanner.datastore.https;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.zip.GZIPInputStream;
+import org.opentripplanner.datastore.api.DataSource;
+import org.opentripplanner.datastore.api.FileType;
+import org.opentripplanner.datastore.file.DirectoryDataSource;
+import org.opentripplanner.datastore.file.ZipFileDataSource;
+import org.opentripplanner.util.HttpUtils;
+
+/**
+ * This class is a wrapper around an HTTPS resource.
+ * <p>
+ * Reading compressed HTTPS resources is supported. The only format supported is gzip (extension
+ * .gz).
+ */
+record HttpsFileDataSource(URI uri, FileType type) implements DataSource {
+  /**
+   * Create a data source wrapper around an HTTPS resource. This wrapper handles GZIP(.gz)
+   * compressed files as well as normal files. It does not handle
+   * directories({@link DirectoryDataSource}) or zip-files {@link ZipFileDataSource} which contain
+   * multiple files.
+   */
+
+  @Override
+  public long size() {
+    return -1;
+  }
+
+  @Override
+  public long lastModified() {
+    return -1;
+  }
+
+  @Override
+  public boolean exists() {
+    return true;
+  }
+
+  @Override
+  public boolean isWritable() {
+    return false;
+  }
+
+  @Override
+  public InputStream asInputStream() {
+    InputStream in;
+
+    try {
+      in = HttpUtils.getData(uri);
+    } catch (IOException e) {
+      throw new IllegalStateException(e.getLocalizedMessage(), e);
+    }
+
+    // We support both gzip and unzipped files when reading.
+
+    if (uri.getPath().endsWith(".gz")) {
```

Comment:
Should we do a HEAD request and check the returned mime type instead? That way we could also verify that the resource exists

Reply:
Correct, this test does not work. We should check the Content-Encoding header instead:
Content-Encoding: gzip

Reply:
The content encoding is used just for the in-flight payload, and the extraction should be handled by the HTTP library

Reply:
You're right, the check should be:
.gz extension OR content-type="application/gzip"